src/control/destination/background/mod.rs

```diff
@@ -67,9 +67,9 @@ struct Background<T: HttpService<ResponseBody = RecvBody>> {
 /// which require reconnects.
 #[derive(Default)]
 struct DestinationCache<T: HttpService<ResponseBody = RecvBody>> {
-    destinations: HashMap<DnsNameAndPort, DestinationSet<T>>,
+    destinations: HashMap<Arc<DnsNameAndPort>, DestinationSet<T>>,
     /// A queue of authorities that need to be reconnected.
-    reconnects: VecDeque<DnsNameAndPort>,
+    reconnects: VecDeque<Weak<DnsNameAndPort>>,
```

Comment:
Is it really necessary to make this change -- putting the names on the heap -- to solve the problem of passing the names around?